pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -208,17 +208,17 @@ func (s *DiscoveryServer) ClearCacheFunc() func() {
 			// TODO: metric !!
 			return
 		}
+		s.env.PushContext = push
+		versionLocal := time.Now().Format(time.RFC3339)
 		initContextTime := time.Since(t0)
+		adsLog.Infof("InitContext %v for push takes %s", versionLocal, initContextTime)
 
 		// TODO: propagate K8S version and use it instead
 		versionMutex.Lock()
-		s.env.PushContext = push
```

Comment:
@costinm was the mutex intended to guard these updates as well?

Reply:
Not indeed, `s.env.PushContext` is not locked all over this repo.